lib/tasks/omdb.rake

```diff
@@ -0,0 +1,250 @@
+      'tt2582802',  'tt0078748',  'tt0482571', 'tt0407887', 'tt0088763',
+      'tt0110413',  'tt1853728',  'tt0120586', 'tt0114814', 'tt0110357',
+      'tt0253474',  'tt0034583',  'tt1675434', 'tt0047396', 'tt0054215',
+      'tt0095327',  'tt0064116',  'tt0095765', 'tt0021749', 'tt0027977',
+      'tt15398776', 'tt7286456',  'tt0361748', 'tt4154796', 'tt0090605',
+      'tt1345836',  'tt0209144',  'tt4633694', 'tt0082971', 'tt0078788',
+      'tt0081505',  'tt0086879',  'tt0091251', 'tt4154756', 'tt5311514',
+      'tt1187043',  'tt2380307',  'tt0910970', 'tt0405094', 'tt0057012',
+      'tt8267604',  'tt0043014',  'tt0082096', 'tt0051201', 'tt0050825',
+      'tt0032553',  'tt0057565',  'tt0364569', 'tt0119217', 'tt0180093',
+      'tt0169547',  'tt0338013',  'tt0112573', 'tt2106476', 'tt0062622',
+      'tt0105236',  'tt0087843',  'tt0086190', 'tt0114709', 'tt0119698',
+      'tt0045152',  'tt0056172',  'tt0435761', 'tt0053604', 'tt0044741'
+    ]
+
+    ActiveRecord::Base.transaction do
+      top_100_movies_imdb_id.each do |imdb_id|
+        puts "Searching imdb id:#{imdb_id} using OMDB api..."
+        response = fetch_movie_details(imdb_id)
+
+        # Extracting movie details from the API response
+        actors = response['Actors'].split(', ')
+        awards = response['Awards']
+        box_office = response['BoxOffice']
+        countries = response['Country'].split(', ')
+        directors = response['Director'].split(', ')
+        dvd_release = response['DVD']
+        genres = response['Genre'].split(', ')
+        imdb_rating = response['imdbRating']
+        imdb_votes = response['imdbVotes']
+        imdb_id = response['imdbID']
+        languages = response['Language'].split(', ')
+        metascore = response['Metascore']
+        plot = response['Plot']
+        production = response['Production']
+        poster = response['Poster']
+        rated = response['Rated']
+        ratings = response['Ratings']
+        released = response['Released']
+        runtime = response['Runtime']
+        title = response['Title']
+        website = response['Website']
+        writers = response['Writer'].split(', ')
+        year = response['Year']
+
+        # Creating a new Movie object and saving it to the database
+        movie = Movie.new
+
+        movie.awards = awards
+        movie.box_office = box_office
+        movie.dvd_release = dvd_release
+        movie.imdb_rating = imdb_rating
+        movie.imdb_votes = imdb_votes
+        movie.id = imdb_id
+        movie.metascore = metascore
+        movie.plot = plot
+        movie.production = production
+        movie.poster = poster
+        movie.rated = rated
+        movie.released = released
+        movie.runtime = runtime
+        movie.title = title
+        movie.website = website
+        movie.year = year
+
+        # Calling methods to save related details to the database
+        save_actors(actors, imdb_id)
```

Comment:
Try to use `accepts_nested_attributes_for`

https://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html

Reply:
same for other associations